monkestation/code/modules/antagonists/contractor/items/modsuit/modules.dm

```diff
@@ -0,0 +1,87 @@
+/obj/item/mod/module/baton_holster
+	name = "MOD baton holster module"
+	desc = "A module installed into the chest of a MODSuit, this allows you \
+		to retrieve an inserted baton from the suit at will. Insert a baton \
+		by using the module with the baton in hand. \
+		Remove an inserted baton by using a wrench on the module while it is removed from the suit."
+	icon_state = "holster"
+	icon = 'monkestation/icons/obj/items/modsuit_modules.dmi'
+	complexity = 3
+	incompatible_modules = list(/obj/item/mod/module/baton_holster)
+	module_type = MODULE_USABLE
+	/// Ref to the baton
+	var/obj/item/melee/baton/telescopic/contractor_baton/stored_batong
+	/// If the baton is out or not
+	var/deployed = FALSE
+
+// doesn't give a shit if it's deployed or not
+/obj/item/mod/module/baton_holster/on_select()
+	on_use()
+	SEND_SIGNAL(mod, COMSIG_MOD_MODULE_SELECTED, src)
+
+/obj/item/mod/module/baton_holster/wrench_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(!stored_batong)
+		return
+	balloon_alert(user, "[stored_batong] removed")
+	stored_batong.forceMove(get_turf(src))
+	stored_batong.holster = null
+	stored_batong = null
+	tool.play_tool_sound(src)
+
+/obj/item/mod/module/baton_holster/Destroy()
+	if(stored_batong)
+		stored_batong.forceMove(get_turf(src))
+		stored_batong.holster = null
+		stored_batong = null
+	. = ..()
```

Comment:
```suggestion
	return ..()
```

Reply:
Dont start review yet, I still have a bunch of things to port, a decent amount of which are refactors.